pages/signinregister.js

```diff
@@ -4,18 +4,23 @@ import Head from '../components/head'
 import LoginForm from '../components/loginform'
 import '../static/styles/main.scss'
 
-const SignInSignUp = () => (
+const SignInSignUp = (props) => (
   <div className='main-container'>
     <Nav navColor='white' />
     <Head title='Sign In' />
 
     <div className='body text-center'>
       <div className='login-container'>
-        <LoginForm />
+        <LoginForm type={props.teacher}/>
       </div>
     </div>
     <Footer />
   </div>
 )
 
+SignInSignUp.getInitialProps = async ({ query }) => {
+  console.log(query.type) // { type: 'teacher' }
+  return { teacher: query.type === 'teacher' }
```

Comment:
So here you could actually just return `{ type: query.type }` and pass that type to the child component.